### PR TITLE
Set icon during runtime

### PR DESCRIPTION
**Pull request by tinolaomahei**
Currently the icon can only be set at design time. These changes allow users to call the `setIcon` method by passing in a drawable resource id e.g.

```java
SlideToActView sta = findViewById(R.id.slider);
sta.setIcon(R.drawable.custom_icon);
```
I've added an example in the example app and have included the new setter in the README file.

Let me know if there's something I've missed or if my code doesn't work :P

### CODE REVIEW

**Comment by quanta-kt** [comment]
In my opinion, you should use `mIcon` field for setting custom icons from drawable resources. As the view might be recreated on the an orientation change and the constructor will change icon back to default if that happens. Refer to line number 343 for more context on this.

+A Suggestion: It would be very great to use Kotlin properties instead of classic Java setter. I'd normally tend to use this syntax in Kotlin:
```kt
slider.icon = R.drawable.custom_icon
```

**Comment by tinolaomahei** [comment]
@quanta-kt I agree! I'll make some adjustments and push soon. Cheers.

File: README.md [hunk 1]
```diff
@@ -220,6 +220,13 @@ You can set a custom icon by setting the ``slider_icon``attribute to a drawable
 app:slider_icon="@drawable/custom_icon"
 ```
 
+You can also set a custom icon programmtically with the provided setter.
```
Comment (cortinico) [inline README.md:223]:
```suggestion
You can also set a custom icon programmatically with the provided setter.
```

File: README.md [hunk 2]
```diff
@@ -220,6 +220,13 @@ You can set a custom icon by setting the ``slider_icon``attribute to a drawable
...
+
+```java
+SlideToActView sta = findViewById(R.id.slider);
+sta.setIcon(R.drawable.custom_icon);
```
Comment (cortinico) [inline README.md:227]:
```suggestion
sta.setSliderIcon(R.drawable.custom_icon);
```

File: example/src/main/java/com/ncorti/slidetoact/example/SampleActivity.java [hunk 3]
```diff
@@ -65,6 +65,36 @@ protected void onCreate(Bundle savedInstanceState) {
             case R.id.button_custom_icon:
                 setContentView(R.layout.content_custom_icon);
                 break;
+            case R.id.button_custom_icon_runtime:
```
Comment (cortinico) [inline example/src/main/java/com/ncorti/slidetoact/example/SampleActivity.java:68]: Can I ask you to have everything inside a single `custom_icon` button?
Maybe we can have a section below the XML approach with the UI you proposed 👍

File: slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt [hunk 4]
```diff
@@ -141,6 +141,15 @@ class SlideToActView @JvmOverloads constructor(
             invalidate()
         }
 
+    /** Custom Icon */
+    var icon: Int = R.drawable.slidetoact_ic_arrow
```
Comment (cortinico) [inline slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt:145]:
```suggestion
    var sliderIcon: Int = R.drawable.slidetoact_ic_arrow
```

That's basically to:

- Be consistent with the `slider_icon` XML property
- Allow for future expansion for a potential `tick_icon`/`tickIcon` (see #14)

File: slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt [hunk 5]
```diff
@@ -141,6 +141,15 @@ class SlideToActView @JvmOverloads constructor(
             invalidate()
         }
 
+    /** Custom Icon */
```
Comment (cortinico) [inline slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt:144]:
```suggestion
    /** Custom Slider Icon */
```

File: slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt [hunk 6]
```diff
@@ -141,6 +141,15 @@ class SlideToActView @JvmOverloads constructor(
...
+    var icon: Int = R.drawable.slidetoact_ic_arrow
+        set(value) {
```
Comment (cortinico) [inline slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt:146]: I'm afraid this custom setter is causing a parsing of `VectorDrawable` twice (once for the field initialiser and once again during ctoring).

Ideally we can change it with something like:
```kotlin
/** Custom Slider Icon */
    var sliderIcon : Int = 0
        set(value) {
            field = value
            if (field != 0) {
                mDrawableArrow = parseVectorDrawableCompat(context.resources, value, context.theme)
                mDrawableArrow.setTint(iconColor)
                invalidate()
            }
        }
```

File: README.md (hunk 1 above)
Comment (tinolaomahei) [reply README.md]: Oops, typing too fast :P

File: README.md (hunk 2 above)
Comment (tinolaomahei) [reply README.md]: Makes sense. Have included in the latest commit.

File: example/src/main/java/com/ncorti/slidetoact/example/SampleActivity.java (hunk 3 above)
Comment (tinolaomahei) [reply example/src/main/java/com/ncorti/slidetoact/example/SampleActivity.java]: Perhaps we don't need to have two sliders in the custom icon example. Checkout my latest commit, I'm using the same slider to demonstrate both methods of setting custom icons. Let me know what you think.

I've removed the "custom icon at runtime" views and references.

File: slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt (hunk 4 above)
Comment (tinolaomahei) [reply slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt]: Have included this change in my latest commit.

File: slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt (hunk 5 above)
Comment (tinolaomahei) [reply slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt]: Naturally :)

File: slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt (hunk 6 above)
Comment (tinolaomahei) [reply slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt:146]: Good catch. I've added the check in my latest commit.

**Pull request merged**